Notify/Notify/Notify/AppShell.xaml.cs

```diff
@@ -106,82 +100,70 @@ private void setMessagingCenterStopServiceMessageSubscription()
                     }
                     catch (Exception ex)
                     {
-                        Debug.Write("Failed in MessagingCenter.Subscribe<StopServiceMessage>: " + ex.Message);
+                        Debug.Write($"Failed in MessagingCenter.Subscribe<StopServiceMessage>: {ex.Message}");
                     }
                 });
             });
         }
 
-        private void setMessagingCenterLocationMessageSubscription()
+        private void setMessagingCenterLocationSubscription()
         {
-            MessagingCenter.Subscribe<LocationMessage>(this, "Location", location =>
+            MessagingCenter.Subscribe<Location>(this, "Location", location =>
             {
-                if (currentLocation != location)
+                bool arrived;
+
+                if (isCurrentLocationFarEnoughForUpdating(location))
                 {
-                    currentLocation = location;
-                    Debug.WriteLine($"{location.Latitude}, {location.Longitude}, {DateTime.Now.ToLongTimeString()}");
-                    double distance = sendCurrentLocationToAzureFunction(location).Result;
-                    if (checkIfArrivedDestinationForTheFirstTime(distance))
+                    Debug.WriteLine($"{location}, {DateTime.Now.ToLongTimeString()}");
+                    arrived = AzureHttpClient.Instance.CheckIfArrivedDestination(location);
+                    
+                    if (arrived)
                     {
                         Device.BeginInvokeOnMainThread(() =>
                         {
                             try
                             {
-                                arrivedDestinationForTheFirstTime();
+                                AnnounceDestinationArrival();
                             }
                             catch (Exception ex)
                             {
-                                Debug.WriteLine("Failed in MessagingCenter.Subscribe<LocationMessage>: " + ex.Message);
+                                Debug.WriteLine($"Failed in MessagingCenter.Subscribe<Location>: {ex.Message}");
                             }
                         });
                     }
+                    
+                    m_LastUpdatedLocation = location;
+                    Debug.WriteLine($"Updated last updated location: {m_LastUpdatedLocation}");
                 }
             });
         }
         
-        private async Task<double> sendCurrentLocationToAzureFunction(LocationMessage currentLocation)
+        bool isCurrentLocationFarEnoughForUpdating(Location location)
```

Comment:
Consider changing the method name to something like : `requiresLocationUpdate` so it conveys the idea that the method determines whether the current location requires a location update based on the distance threshold.